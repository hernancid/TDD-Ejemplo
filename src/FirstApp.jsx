import PropTypes from 'prop-types';
export const FirstApp = ({title, subTitle}) => {
    return (
        <>
            <h1>{title}</h1>
            <p>{subTitle}</p>
        </>
    );
}

FirstApp.propTypes = {
    title: PropTypes.string.isRequired
}
FirstApp.defaultProps = {
    title: 'No Hay titulo',
    subTitle: 'No hay subtitulo'
}