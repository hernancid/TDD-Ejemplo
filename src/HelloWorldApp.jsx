export const HelloWorldApp= () =>{
    return (
        <h1>Holi</h1>
    );
}