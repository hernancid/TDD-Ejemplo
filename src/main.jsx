import React from 'react';
import ReactDOM from 'react-dom/client';

import {HelloWorldApp} from './HelloWorldApp.jsx';
import {FirstApp} from "./FirstApp.jsx";
import {CounterApp} from "./CounterApp.jsx";
import './style.css'
ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
        <FirstApp title="Hola, Soy Vegeta"/>
        <HelloWorldApp/>
        <CounterApp value={0}/>
    </React.StrictMode>
)

