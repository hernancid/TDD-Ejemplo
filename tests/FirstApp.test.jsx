import {getAllByAltText, getByTestId, render, screen} from "@testing-library/react";
import {FirstApp} from "../src/FirstApp";

describe('Pruebes en FirstApp.jsx',()=>{
    const title = 'Hola, Soy Goku';
    const subTitle = 'Soy un subtitulo';
    test('Debe de hacer match con el snapshot', ()=>{
        const {container} = render(<FirstApp title={title}/>);
        expect(container).toMatchSnapshot();
    });

    test('Debe de mostrar el mensaje "Hola, Soy Goku"',()=>{
        render(<FirstApp title={title}/>);
        expect(screen.getByText(title)).toBeTruthy();
    });

    test('debe mostrar el titulo en un h1', ()=>{
        render(<FirstApp title={title}/>);
        expect(screen.getByRole('heading', {level: 1}).innerHTML).toContain(title);
    })

    test('debe de mostrar el subtitulo enviado por las props', ()=>{
        render(<FirstApp title={title} subTitle={subTitle}/>);
        expect(screen.getAllByText(subTitle).length).toBe(1);

    })
})