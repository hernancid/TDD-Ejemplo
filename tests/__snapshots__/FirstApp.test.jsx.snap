// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebes en FirstApp.jsx Debe de hacer match con el snapshot 1`] = `
<div>
  <h1>
    Hola, Soy Goku
  </h1>
  <p>
    No hay subtitulo
  </p>
</div>
`;
